### PR TITLE
unhide-gui: init at 20220611

### PR DESCRIPTION
**Pull request by Tochiaha**
Releases: https://github.com/YJesus/Unhide/releases/tag/v20220611

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [x] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Tochiaha** [comment]
@superherointj

File: pkgs/by-name/un/unhideGui/package.nix [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  buildInputs = [ unhide ];
+
+  postPatch = ''
+    substituteInPlace unhideGui.py --replace-fail "\This" "This"
+    substituteInPlace unhideGui.py --replace-fail "__credits__" "#__credits__"
+    substituteInPlace unhideGui.py --replace-fail "./unhide-linux" "${unhide}/bin/unhide-linux"
+    substituteInPlace unhideGui.py --replace-fail "./unhide-tcp" "${unhide}/bin/unhide-tcp"
+  '';
+
+  installPhase = ''
```
Comment (superherointj) [inline pkgs/by-name/un/unhideGui/package.nix:32]: missing hooks

File: pkgs/by-name/un/unhideGui/package.nix [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+  propagatedBuildInputs = with python3Packages; [ tkinter ];
+
+  buildInputs = [ unhide ];
+
+  postPatch = ''
+    substituteInPlace unhideGui.py --replace-fail "\This" "This"
+    substituteInPlace unhideGui.py --replace-fail "__credits__" "#__credits__"
+    substituteInPlace unhideGui.py --replace-fail "./unhide-linux" "${unhide}/bin/unhide-linux"
+    substituteInPlace unhideGui.py --replace-fail "./unhide-tcp" "${unhide}/bin/unhide-tcp"
```
Comment (superherointj) [inline pkgs/by-name/un/unhideGui/package.nix:29]:
```suggestion
    substituteInPlace unhideGui.py \
      --replace-fail "\This" "This" \
      --replace-fail "__credits__" "#__credits__" \
      --replace-fail "./unhide-linux" "${unhide}/bin/unhide-linux" \
      --replace-fail "./unhide-tcp" "${unhide}/bin/unhide-tcp"
```

**Comment by superherointj** [comment]
Regarding package name, please read:

https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-naming

> The pname and the version attribute must not contain uppercase letters — e.g., "mplayer" instead of "MPlayer".

File: pkgs/by-name/un/unhide-gui/package.nix [hunk 3]
```diff
@@ -0,0 +1,61 @@
+{
+  fetchFromGitHub,
+  lib,
+  python3Packages,
+  python3,
+  unhide,
+}:
+
+python3Packages.buildPythonApplication rec {
+  pname = "unhide";
```
Comment (superherointj) [inline pkgs/by-name/un/unhide-gui/package.nix:10]:
```suggestion
  pname = "unhide-gui";
```
It should match top-level.

**Comment by superherointj** [comment]
Please, squash commits.

File: pkgs/by-name/un/unhide-gui/package.nix [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+    makeWrapper "${python3Packages.python.interpreter}" "$out/bin/unhideGui" \
+      --set PYTHONPATH "$PYTHONPATH" \
+      --add-flags "$out/share/unhideGui/unhideGui.py"
+
+    runHook postFixup
+  '';
+
+  meta = {
+    description = "Stable version of Unhide";
```
Comment (superherointj) [inline pkgs/by-name/un/unhide-gui/package.nix:53]:
```suggestion
    description = "Forensic tool to find hidden processes and TCP/UDP ports by rootkits, LKMs or other hiding technique";
```

**Comment by superherointj** [comment]
> Please, (always) squash commits.

^

**Comment by Tochiaha** [comment]
squashed

**Review by superherointj — APPROVED** [review]

**Pull request merged**